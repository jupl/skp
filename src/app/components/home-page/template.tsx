import * as React from 'react'
import {Container, Header} from 'semantic-ui-react'

/**
 * Render home page component
 * @return Home page component
 */
export function AppHomePage() {
  return (
    <Container>
      <Header>Hello, world</Header>
    </Container>
  )
}
