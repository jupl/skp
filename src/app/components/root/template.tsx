import * as React from 'react'
import {Container, Header, Menu} from 'semantic-ui-react'
import styled from 'styled-components'

// Container component
const Layout = styled.div`
  width: 100vw;
  height: 100vh;
  overflow: hidden;
  display: flex;
  flex-direction: column;
`
const MenuBar = styled.div``
const Body = styled.div`
  flex: 1;
  overflow: auto;
`

/**
 * Render root component representing the entire application
 * @return Root component
 */
export function AppRoot() {
  return (
    <Layout>
      <MenuBar>
        <Container>
          <Menu secondary>
            <Menu.Item header>Application</Menu.Item>
          </Menu>
        </Container>
      </MenuBar>
      <Body>
        <Container>
          <Header>Hello, world</Header>
        </Container>
      </Body>
    </Layout>
  )
}
