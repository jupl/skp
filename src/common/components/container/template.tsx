import * as React from 'react'
import {ApolloProvider} from 'react-apollo'
import {ProviderProps} from 'react-apollo/ApolloProvider'
import {AppContainer} from 'react-hot-loader'

/** Container component properties */
interface Props<P> extends ProviderProps {
  /** Component to render */
  readonly component: React.ComponentClass<P> | (() => React.ReactElement<P>)
}

/**
 * Render container component adding possible dev tools and redux store
 * @param props Component properties
 * @return Container component
 */
export function Container<P>({component: Component, ...props}: Props<P>) {
  return (
    <AppContainer>
      <ApolloProvider {...props}>
        <Component />
      </ApolloProvider>
    </AppContainer>
  )
}
