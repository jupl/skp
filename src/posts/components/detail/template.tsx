import * as React from 'react'
import {Post} from '../../graphql'

/** Interfaces for PostsList component properties */
export namespace PostsDetail {
  /** Component data */
  export interface Data {
    post: Post
  }
  /** Component properties */
  export type Props = Data
}

/** Render list of posts */
export function PostsDetail({post}: PostsDetail.Props) {
  return <div>{post.title}</div>
}
