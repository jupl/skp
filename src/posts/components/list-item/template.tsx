import * as React from 'react'
import {Post} from '../../graphql'

/** Interfaces for PostsListItem component properties */
export namespace PostsListItem {
  /** Component data */
  export interface Data {
    post: Post
  }
  /** Component actions */
  export interface Actions {
    onClick: React.EventHandler<React.MouseEvent<HTMLDivElement>>
  }
  /** Component properties */
  export type Props = Actions & Data
}

/** Render posts list item */
export function PostsListItem({post, ...props}: PostsListItem.Props) {
  return <div {...props}>{post.title}</div>
}
