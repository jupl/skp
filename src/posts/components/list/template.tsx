import * as React from 'react'
import {Post} from '../../graphql'
import {PostsListItem} from '../list-item'

/** Interfaces for PostsList component properties */
export namespace PostsList {
  /** Component data */
  export interface Data {
    posts: Post[]
  }
  /** Component actions */
  export interface Actions {
    selectPost(id: Post['id']): void
  }
  /** Component properties */
  export type Props = Data & Actions
}

/** Render list of posts */
export function PostsList({posts, selectPost}: PostsList.Props) {
  return <div>{posts.map(createPostsListItem)}</div>

  function createPostsListItem(post: Post) {
    return <PostsListItem key={post.id} post={post} onClick={onClick} />

    function onClick() {
      selectPost(post.id)
    }
  }
}
