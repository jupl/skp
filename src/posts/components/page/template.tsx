import * as React from 'react'
import {Container, Header} from 'semantic-ui-react'
import {PostsList} from '../list'

/**
 * Render home page component
 * @return Home page component
 */
export function PostsPage() {
  return (
    <Container>
      <Header>Hello, posts</Header>
      <PostsList />
    </Container>
  )
}
