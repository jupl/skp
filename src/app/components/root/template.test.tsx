import {shallow} from 'enzyme'
import * as React from 'react'
import {AppRoot} from './template'

describe('<AppRoot> Template', () => {
  it('should render as expected', () => {
    expect(shallow(<AppRoot />)).toMatchSnapshot()
  })
})
