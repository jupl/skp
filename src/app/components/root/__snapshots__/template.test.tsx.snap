// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRoot> Template should render as expected 1`] = `
<styled.div>
  <styled.div>
    <Container>
      <Menu
        secondary={true}
      >
        <MenuItem
          header={true}
        >
          Application
        </MenuItem>
      </Menu>
    </Container>
  </styled.div>
  <styled.div>
    <Container>
      <Header>
        Hello, world
      </Header>
    </Container>
  </styled.div>
</styled.div>
`;
